control-plane-operator/controllers/hostedcontrolplane/configoperator/params.go

```diff
@@ -43,6 +44,15 @@ func NewHostedClusterConfigOperatorParams(ctx context.Context, hcp *hyperv1.Host
 			},
 		},
 	}
+
+	if explicitNonRootSecurityContext {
```

Comment:
Do we ever have a container that should not have the RunAsUser? If not, we should probably instead just iterate over all of them, this also has the advantage that it will do the right thing in the future if we add additional containers